RecoMuon/MuonIdentification/src/MuonTimingFiller.cc

```diff
@@ -44,11 +45,21 @@ MuonTimingFiller::MuonTimingFiller(const edm::ParameterSet& iConfig, edm::Consum
 {
    // Load parameters for the DTTimingExtractor
    edm::ParameterSet dtTimingParameters = iConfig.getParameter<edm::ParameterSet>("DTTimingParameters");
-   theDTTimingExtractor_ = new DTTimingExtractor(dtTimingParameters,iC);
 
    // Load parameters for the CSCTimingExtractor
    edm::ParameterSet cscTimingParameters = iConfig.getParameter<edm::ParameterSet>("CSCTimingParameters");
-   theCSCTimingExtractor_ = new CSCTimingExtractor(cscTimingParameters,iC);
+
+   // Fallback mechanism for old configs (there the segment matcher was built inside the timing extractors)
+   edm::ParameterSet matchParameters;
+   if (iConfig.existsAs<edm::ParameterSet>("MatchParameters")) 
+       matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
+     else 
+       matchParameters = dtTimingParameters.getParameter<edm::ParameterSet>("MatchParameters");
+   
+   
+   theMatcher_ = new MuonSegmentMatcher(matchParameters, iC);
```

Comment:
it could be a good opportunity to switch to unique_ptr to let it handle the lifetime of the object and drop the delete